qln/dlc.go

```diff
@@ -0,0 +1,611 @@
+package qln
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+
+	"github.com/adiabat/btcd/btcec"
+	"github.com/adiabat/btcd/txscript"
+	"github.com/adiabat/btcd/wire"
+	"github.com/adiabat/btcutil/txsort"
+	"github.com/mit-dci/lit/lnutil"
+	"github.com/mit-dci/lit/portxo"
+	"github.com/mit-dci/lit/sig64"
+)
+
+func (nd *LitNode) AddContract() (*lnutil.DlcContract, error) {
+
+	c, err := nd.DlcManager.AddContract()
+	if err != nil {
+		return nil, err
+	}
+
+	return c, nil
+}
+
+func (nd *LitNode) OfferDlc(peerIdx uint32, cIdx uint64) error {
+	c, err := nd.DlcManager.LoadContract(cIdx)
+	if err != nil {
+		return err
+	}
+
+	if c.Status != lnutil.ContractStatusDraft {
+		return fmt.Errorf("You cannot offer a contract to someone that is not in draft stage")
+	}
+
+	if !nd.ConnectedToPeer(peerIdx) {
+		return fmt.Errorf("You are not connected to peer %d, do that first", peerIdx)
+	}
+
+	var nullBytes [33]byte
+	// Check if everything's set
+	if bytes.Equal(nullBytes[:], c.OracleA[:]) {
+		return fmt.Errorf("You need to set an oracle for the contract before offering it")
+	}
+
+	if bytes.Equal(nullBytes[:], c.OracleR[:]) {
+		return fmt.Errorf("You need to set an R-point for the contract before offering it")
+	}
+
+	if c.OracleTimestamp == 0 {
+		return fmt.Errorf("You need to set a settlement time for the contract before offering it")
+	}
+
+	if c.CoinType == 0 {
```

Comment:
Added a new constant COINTYPE_NOT_SET with ^uint32(0) (maximum UINT32) as value. I think we won't have that many coins eventually, so this will probably be a safe value to use.